src/http/client.cc

```diff
@@ -171,11 +171,15 @@ input_stream<char> connection::in(reply& rep) {
 }
 
 future<> connection::close() {
-    return when_all(_read_buf.close(), _write_buf.close()).discard_result().then([this] {
-        auto la = _fd.local_address();
-        return std::move(_closed).then([la = std::move(la)] {
-            http_log.trace("destroyed connection {}", la);
-        });
+    return when_all(
+        _read_buf.close().handle_exception([] (auto ignored) {}),
+        _write_buf.close().handle_exception([] (auto ignored) {}))
```

Comment:
That's the only place that can produce exception, isn't it? I mean that `input_stream::close()` is not resolving into exceptions. (#1150)

In fact, we're trying to make all close()-s not to resolve into exceptions, so it probably should be output_stream::close()'s fix, not http client's, all the more so rpc clients also close write streams without checking for exceptions

Cc @bhalevy , WDYT?

Reply:
We definitely want read resource close to never return an error.
Historically, for files, we typically do flush in close, and flush may fail, but it's still the caller's responsibility to invoke flush/fsync before close if it is interested in its completion status, so I think it's OK to handle all exceptions in close and log a warning when ignoring them.

@avikivity are you okay with that?